framework/src/positions/QuadraturePointsPositions.C

```diff
@@ -0,0 +1,92 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "QuadraturePointsPositions.h"
+#include "libmesh/quadrature_gauss.h"
+
+registerMooseObject("MooseApp", QuadraturePointsPositions);
+
+InputParameters
+QuadraturePointsPositions::validParams()
+{
+  InputParameters params = Positions::validParams();
+  params.addClassDescription("Positions of element quadrature points. Only supports the default "
+                             "Gauss quadrature with default order elements");
+  params += BlockRestrictable::validParams();
+
+  // Element centroids could be sorted by XYZ or by id. Default to not sorting
+  params.set<bool>("auto_sort") = false;
+  // Gathered locally, should be broadcast on every process
+  params.set<bool>("auto_broadcast") = true;
+
+  return params;
+}
+
+QuadraturePointsPositions::QuadraturePointsPositions(const InputParameters & parameters)
+  : Positions(parameters), BlockRestrictable(this), _mesh(_fe_problem.mesh())
+{
+  // Mesh is ready at construction
+  initialize();
+  // Trigger synchronization as the initialization is distributed
+  finalize();
+}
+
+void
+QuadraturePointsPositions::initialize()
+{
+  clearPositions();
+
+  // By default, initialize should be called on meshChanged()
+  // Gathering of positions is local, reporter system makes sure to make it global
+  if (blockRestricted())
+  {
+    _positions_2d.resize(numBlocks());
+    unsigned int b_index = 0;
+    for (const auto & sub_id : blockIDs())
+    {
+      for (const auto & elem : _mesh.getMesh().active_local_subdomain_elements_ptr_range(sub_id))
+      {
+        // Get a quadrature going on the element
+        const FEFamily mapping_family = FEMap::map_fe_type(*elem);
+        const FEType fe_type(elem->default_order(), mapping_family);
+        std::unique_ptr<FEBase> fe = FEBase::build(elem->dim(), fe_type);
+        QGauss qrule(elem->dim(), fe_type.default_quadrature_order());
```

Comment:
I feel like a user would more naturally expect this to match what’s used in Assembly which is controllable using the input file where as this hard coding is not

Reply:
ok changed. matching assembly comes at a cost (cant run as early, add_positions task is very early) though so I did not do exactly that